overlord/snapstate/handlers.go

```diff
@@ -1907,7 +1907,9 @@ func (m *SnapManager) undoStartSnapServices(t *state.Task, _ *tomb.Tomb) error {
 	var stopReason snap.ServiceStopReason
 
 	// stop the services
+	st.Unlock()
```

Comment:
I think it'd be useful to review other places where similarly long-running backend calls are made.